src/rest/AnalyticsEndpoint.ts

```diff
@@ -93,22 +95,40 @@ export class AnalyticsEndpoint {
     return this.getFromService<string[]>(url, params);
   }
 
+  public forgetVisitorId() {
+    Cookie.erase('visitorId');
+  }
+
+  public kill() {
+    this.dead = true;
+  }
+
   private async sendToService(data: Record<string, any>, path: string, paramName: string): Promise<any> {
     // We use pendingRequest because we don't want to have 2 request to analytics at the same time.
     // Otherwise the cookie visitId won't be set correctly.
     if (AnalyticsEndpoint.pendingRequest != null) {
       await AnalyticsEndpoint.pendingRequest;
     }
+    if (this.dead) {
+      return;
+    }
 
     const url = this.getURL(path);
     const request = this.executeRequest(url, data);
 
     try {
       const results = await request;
+      if (this.dead) {
```

Comment:
This seems un-necessary here ? Since we return early in the function anyway if this is true.

Reply:
I added it there because it could've passed prior to awaiting the request, but this was unnecessary caution.

This is resolved as I removed the ability to disable [AnalyticsEndpoint](../tree/master/src/rest/AnalyticsEndpoint.ts) directly.